In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
from sklearn.model_selection import train_test_split #imports the train_test_split function from scikit-learn’s model_selection module
from sklearn.preprocessing import StandardScaler #loading in train_test_split function from scikit-learn’s model_selection module
from sklearn.linear_model import Perceptron #loading in train_test_split function from scikit-learn’s model_selection module
from sklearn.linear_model import LogisticRegression #loading in train_test_split function from scikit-learn’s model_selection module
from sklearn.tree import DecisionTreeClassifier #loading in train_test_split function from scikit-learn’s model_selection module
from sklearn.ensemble import RandomForestClassifier #loading in train_test_split function from scikit-learn’s model_selection module
from sklearn.neighbors import KNeighborsClassifier #loading in train_test_split function from scikit-learn’s model_selection module
from sklearn.metrics import accuracy_score, classification_report #loading in train_test_split function from scikit-learn’s model_selection module



# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/forest/train.csv
/kaggle/input/forest/test.csv


In [2]:
train = pd.read_csv('/kaggle/input/forest/train.csv') #import train data set as "mydata"

**Part 2: Preprocessing and Feature Engineering**

We would use the "drop" function to drop the "id" column and set the axis to 1 and renaming the new dataset to "train2".

In [3]:
train2 = train.drop('id', axis=1)
train2.head()

Elevation  Aspect  Slope  Horizontal_Distance_To_Hydrology  \
0       3031      38      8                               192   
1       2717      66     14                               384   
2       3172      24      9                               300   
3       3089      67      2                               285   
4       2569      22     20                               216   

   Vertical_Distance_To_Hydrology  Horizontal_Distance_To_Roadways  \
0                              40                               30   
1                             -19                              927   
2                              27                             2389   
3                             -37                             2424   
4                              87                             1095   

   Hillshade_9am  Hillshade_Noon  Hillshade_3pm  \
0            220             223            138   
1            233             211            108   
2            215             220            141   
3            221             235            151   
4            204             194            121   

   Horizontal_Distance_To_Fire_Points  ...  Soil_Type32  Soil_Type33  \
0                                1176  ...            1            0   
1                                2452  ...            0            0   
2                                2408  ...            0            0   
3                                1745  ...            0            0   
4                                1506  ...            0            0   

   Soil_Type34  Soil_Type35  Soil_Type36  Soil_Type37  Soil_Type38  \
0            0            0            0            0            0   
1            0            0            0            0            0   
2            0            0            0            0            0   
3            0            0            0            0            0   
4            0            0            0            0            0   

   Soil_Type39  Soil_Type40  label  
0            0            0      2  
1            0            0      2  
2            0            0      1  
3            0            0      2  
4            0            0      2  

[5 rows x 55 columns]

Next we need to create a new data set with the independent variables called "X" and then "Y" would be the target variable or prediction output.

In [4]:
X = train2.drop('label',axis=1) #independent variables
y = train2['label'] #output

Now we are spliting the datset into 75% training and 25% testing.


In [5]:
X_train, X_test, y_train, y_test = train_test_split(X, y,test_size=0.25, random_state=1) #spliting dataset

We need to ensure that both the training and test datasets are scaled in the same way, so the machine learning algorithms will perform effectively. 

In [6]:
sc = StandardScaler().fit(X_train) #StandardScaler computes the mean and standard deviation for each feature in "X_train"
Xtrain_stnd=sc.transform(X_train) #transforms (scales) the training data
Xtest_stnd=sc.transform(X_test) #transforms (scales) the test data X_test 

**Part 3: Model Training and Evaluation:**

Here we building and using different models such as perceptron, logistic regression, support vector machines (SVM), decision tree, k-nearest neighbors (KNN), and random forest to evaluate each accuracy of the predictions. We are going to use the training and testing data set "Xtrain_stnd" and "Xtest_stnd" we just created.

In [8]:
#Perceptron model

precept = Perceptron(eta0=0.1, random_state=1).fit(Xtrain_stnd, y_train) #building perceptron model
ypre_pred = precept.predict(Xtest_stnd) #testing the model

accuracy = accuracy_score(y_test, ypre_pred) #calculating accuracy 
print("Accuracy:", accuracy)

Accuracy: 0.6242265690214538


In [9]:
#Decision Tree model

dectree = DecisionTreeClassifier(criterion='gini', max_depth=5, random_state=1).fit(Xtrain_stnd, y_train) #building Decision Treee model
yddectree_pred = dectree.predict(Xtest_stnd) #testing the model

accuracy = accuracy_score(y_test, yddectree_pred) #calculating accuracy 
print("Accuracy:", accuracy)

Accuracy: 0.7029508704594546


In [10]:
#K-nearest neighbors (KNN)

knn = KNeighborsClassifier(n_neighbors=5, p=1, metric='euclidean').fit(Xtrain_stnd, y_train) #building K-nearest neighbors (KNN)
yknn_pred = knn.predict(Xtest_stnd) #testing the model

accuracy = accuracy_score(y_test, yknn_pred) #calculating accuracy 
print("Accuracy:", accuracy)

Accuracy: 0.9209228677400756


In [11]:
#Random Forest model

rfc = RandomForestClassifier(n_estimators=100, random_state=1, n_jobs=2).fit(Xtrain_stnd, y_train)  #building Random Forest model
yrfc_pred = rfc.predict(Xtest_stnd) #testing the model

accuracy = accuracy_score(y_test, yrfc_pred) #calculating accuracy 
print("Accuracy:", accuracy)

Accuracy: 0.9503971498154092


In [12]:
#Logistic regression model

lr = LogisticRegression(max_iter=4000, random_state=1).fit(Xtrain_stnd, y_train) #building logistic regression model
ylr_pred = lr.predict(Xtest_stnd) #testing the model

accuracy = accuracy_score(y_test, ylr_pred) #calculating accuracy 
print("Accuracy:", accuracy)

Accuracy: 0.7258590569950862


**Part 4: Model Deployment and Submission**


Now we use the Random Forest model as our final model becuse it produces the highest accuracy of 0.9503971498154092 compared to the other models. We would also need to produce a submission excel sheet with the prediction/outcome of the "test" datset.

In [13]:
test4 = pd.read_csv('/kaggle/input/forest/test.csv') #import test dataset


test6 = test4.drop('id', axis=1) # drop the "id" column


sc3 = StandardScaler().fit(test6)  #StandardScaler computes the mean and standard deviation for each feature in "X_TestEnc"
sc2 = sc3.transform(test6) #transforms (scales) the X_TestEnc data

yrfc2_pred = rfc.predict(sc2) #testing the model

submission_df = pd.DataFrame({ #creating a DataFrame called submission_df with two columns "id" and "label". 
    "id": test4['id'],  
    "label": yrfc2_pred        #"label" contains the outputs/prediction from "yrfc2_pred" made from the Random Forest model
}) 

submission_df.to_csv("submission.csv", index=False) #saves the contents of "submission_df" toa CSV file named "submission.csv" without including he index column